src/debugger/appWorker.ts

```diff
@@ -203,6 +206,9 @@ export class MultipleLifetimesAppWorker {
             if (object.method === "prepareJSRuntime") {
                 // The MultipleLifetimesAppWorker will handle prepareJSRuntime aka create new lifetime
                 this.gotPrepareJSRuntime(object);
+            } else if (object.method === "$disconnected") {
+                // We need to shutdown the current app worker, and create a new lifetime
+                this.singleLifetimeWorker = null;
```

Comment:
We are releasing the global variables in the context. The issue is that our "sandbox" is just switching the global variables, so it's possible that the react-native logic is escaping the sandbox somehow, and then staying alive after we stop using that context.
The solution if that happened would be something like what you two are suggesting, but that would require a complete re-architecture of the debugger.

Still, given that react-native seems to be handling the clean-up somehow, I'd argue to continue with this solution for the time being, until we discover an actual problem that can happen with this current implementation.